src/entity/Ship.java

```diff
@@ -130,6 +190,23 @@ public final boolean isDestroyed() {
         return !this.destructionCooldown.checkFinished();
     }
 
+    /**
+     * 함선이 대각선 방향을 바라보고 있는지 체크
+     *
+     * @return 함선의 방향이 대각선 방향이면 True
+     */
+    public final boolean isDiagonal() {
+        switch (direction) {
+            case UP_RIGHT:
+            case UP_LEFT:
+            case DOWN_RIGHT:
+            case DOWN_LEFT:
+                return true;
+            default:
+                return false;
+        }
```

Comment:
isDiagonal 메소드에서 switch문을 다음처럼 간소화할 수 있을 것 같습니다.
```suggestion
        return switch (direction) {
            case UP_RIGHT, UP_LEFT, DOWN_RIGHT, DOWN_LEFT -> true;
            default -> false;
        };
```